src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -748,81 +754,90 @@ private class BrokerTransaction : IRSessionSwitchBrokerTransaction, IRSessionRec
             }
 
             public async Task CompleteSwitchingBrokerAsync(CancellationToken cancellationToken) {
+                if (_session._disposeToken.IsDisposed) {
```

Comment:
Perhaps make a small method 'bool CheckPrerequisites()` that would check conditions like disposed, null `_lockToken` etc?